src/dso_api/dynamic_api/fields.py

```diff
@@ -145,7 +152,19 @@ class HALLooseRelationUrlField(LooseRelationUrlField):
 
     def to_representation(self, value):
         href = super().to_representation(value)
-        return {"href": href, "title": str(value)}
+        view = self.context["view"]
+        relation = view.model._meta.get_field(to_snake_case(self.field_name)).relation
+        dataset_name, table_name = [to_snake_case(part) for part in relation.split(":")]
+        result = {"href": href, "title": str(value)}
+        try:
+            related_ds = view.table_schema.get_dataset_schema(dataset_name)
+            related_identifier = related_ds.identifier
+            result[related_identifier] = value
+        except AttributeError:
```

Comment:
This should be better addressed in tests, not in field code as it may hide errors in future.

Reply:
And, the try..except should be as brief as possible. If `related_ds.identifier` is missing, the code also fails now.

Reply:
After updating the view in the test, the try-except is not necessary anymore.